openrr-teleop/src/joints_pose_sender.rs

```diff
@@ -0,0 +1,132 @@
+use crate::ControlNode;
+use arci::{
+    gamepad::{Button, GamepadEvent},
+    JointTrajectoryClient, Speaker,
+};
+use async_trait::async_trait;
+use openrr_client::JointsPose;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+pub struct JointsPoseSender<S>
+where
+    S: Speaker,
+{
+    mode: String,
+    joints_poses: Vec<JointsPose>,
+    joint_trajectory_clients: HashMap<String, Arc<dyn JointTrajectoryClient>>,
+    speaker: S,
+    submode: String,
+    pose_index: usize,
+    is_trigger_holding: bool,
+    is_sending: bool,
+    duration: Duration,
+}
+
+impl<S> JointsPoseSender<S>
+where
+    S: Speaker,
+{
+    pub fn new(
+        mode: String,
+        joints_poses: Vec<JointsPose>,
+        all_joint_trajectory_clients: &HashMap<String, Arc<dyn JointTrajectoryClient>>,
```

Comment:
I don't want to introduce the structure of `RobotClient` here.
This should be independent from it as a design.

I think it is not natural to create `JointsPoseSender` with all_joint_trajectory_clients.